robot-db/src/main/java/domain/Book.java

```diff
@@ -93,4 +96,35 @@ public String toString () {
                 .append(description)
                 .append("}").toString();
     }
-}
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(17,31)
+                .append(title)
+                .append(newPrice)
+                .append(oldPrice)
+                .append(currency)
+                .append(year)
+                .append(link)
+                .append(description)
+                .append(printHouse)
+                .append(authors)
+                .append(category)
+                .toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object bookObj) {
+        if (this == bookObj)
+            return true;
+        if (bookObj == null || getClass() != bookObj.getClass())
+            return false;
+        Book book = (Book)bookObj;
+        return title.equals(book.title)
+                && abs(newPrice - book.newPrice) < 1e-10 && abs(oldPrice - book.oldPrice) < 1e-10
+                && currency.equals(book.currency) && year == book.year
+                && link.equals(book.link) && description.equals(book.description)
+                && printHouse.equals(book.printHouse) && authors.equals(book.authors)
+                && category == book.category;
+    }
+}
```

Comment:
I think that for now it's good. We can test it later but it works fast and maybe it's the best solution for this case. I just wanted to know why and now I know that it has sense :+1: